### DISCUSSION TITLE
LangChain OpenAI

### DISCUSSION THREAD

**Discussion by cibernicola** [langflow-ai/langflow#857] [Ideas]
I don't know if this is done on purpose or not, but I'm commenting just in case:

In langchain, all (or much) of what has to do with the OpenAI module, can be configured with own routes and fake api key, which allows seamless integration with models of other types.

Let me explain with an example:

` llm = OpenAI(openai_api_key = "whatever", temperature=0.0, openai_api_base="http://localhost:1234/v1", max_tokens=200).
`
So I think that if langchain "allows" to do it, the langflow "layer" should too, right? 

I insist, I'm sure there are things under de hood that I don't know that can make this not so easy, but I comment it anyway in case it's possible that it can be implemented.

Thanks and good job!

**Comment by moelllerniklas** [comment]
Are you referring to LangFlow Tweaks?

**Reply by cibernicola** [reply]
From the name I would say yes, but I don't know, hence my question.
Do you have more information?

**Comment by carlosrcoelho** [comment]
Hey!

Thanks for sharing your question here. I'm sorry for not getting back to you sooner.
Langflow has gone through some major changes since then. I recommend trying out the new version for the best experience. If you have any issues, please open a new one; we'll be here to help.
